internal/endpoints/endpointsFetcher.go

```diff
@@ -0,0 +1,60 @@
+package endpoints
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+// Schema for data pulled out of EndpointSources file
+type ListOfEndpoints struct {
+	Entries []struct {
+		OrganizationName     string `json:"OrganizationName"`
+		FHIRPatientFacingURI string `json:"FHIRPatientFacingURI"`
+		Type                 string `json:"Type"`
+		Keywords             []struct {
+			Kind  string `json:"Kind"`
+			Value string `json:"Value"`
+		} `json:"Keywords"`
+	} `json:"Entries"`
+}
+
+func GetListOfEndpoints(filePath string) ListOfEndpoints {
+	jsonFile, err := os.Open(filePath)
+	// if we os.Open returns an error then handle it
+	if err != nil {
+		fmt.Println(err)
+	}
+	// defer the closing of our jsonFile so that we can parse it later on
+	defer jsonFile.Close()
```

Comment:
explain defer to me

Reply:
https://gobyexample.com/defer Call the defered function once the function the call is contained in finishes. In this case it makes sure the file doesn't stay open after we are done using it.